src/openrct2/ride/VehiclePaint.cpp

```diff
@@ -1000,63 +998,60 @@ static void vehicle_sprite_paint(
+    paint_session& session, const Vehicle* vehicle, int32_t imageDirection, int32_t z,
+    const rct_ride_entry_vehicle* vehicleEntry)
 {
-    int32_t ecx = ebx / 2;
+    int32_t boundingBoxNum = imageDirection / 2;
     if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_USE_16_ROTATION_FRAMES)
     {
-        ebx = ebx / 2;
+        imageDirection = imageDirection / 2;
     }
     if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_USE_4_ROTATION_FRAMES)
     {
-        ebx = ebx / 8;
+        imageDirection = imageDirection / 8;
     }
-    ebx = (ebx * vehicleEntry->base_num_frames) + vehicle->SwingSprite + vehicleEntry->base_image_id;
-    vehicle_sprite_paint(session, vehicle, ebx, ecx, z, vehicleEntry);
+    auto spriteNum = (imageDirection * vehicleEntry->base_num_frames) + vehicle->SwingSprite + vehicleEntry->base_image_id;
+    vehicle_sprite_paint(
+        session, vehicle, spriteNum, VehicleBoundboxes[vehicleEntry->draw_order][boundingBoxNum], z, vehicleEntry);
 }
 
-// 6D51DE
-static void vehicle_sprite_paint_6D51DE(
-    paint_session& session, const Vehicle* vehicle, int32_t ebx, int32_t z, const rct_ride_entry_vehicle* vehicleEntry)
+static void VehicleSpritePaintRestraints(
+    paint_session& session, const Vehicle* vehicle, int32_t imageDirection, int32_t z,
+    const rct_ride_entry_vehicle* vehicleEntry)
 {
-    if (vehicle->restraints_position < 64)
-    {
-        vehicle_sprite_paint_6D51EB(session, vehicle, ebx, z, vehicleEntry);
-        return;
-    }
-    if (!(vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_RESTRAINT_ANIMATION))
-    {
-        vehicle_sprite_paint_6D51EB(session, vehicle, ebx, z, vehicleEntry);
-        return;
-    }
-    if (ebx & 7)
-    {
-        vehicle_sprite_paint_6D51EB(session, vehicle, ebx, z, vehicleEntry);
-        return;
-    }
-    int32_t ecx = ebx / 2;
-    ebx = ebx / 8;
-    ebx += ((vehicle->restraints_position - 64) / 64) * 4;
-    ebx *= vehicleEntry->base_num_frames;
-    ebx += vehicleEntry->restraint_image_id;
-    vehicle_sprite_paint(session, vehicle, ebx, ecx, z, vehicleEntry);
+    int32_t boundingBoxNum = imageDirection / 2;
+    imageDirection = imageDirection / 8;
+    imageDirection += ((vehicle->restraints_position - 64) / 64) * 4;
+    imageDirection *= vehicleEntry->base_num_frames;
+    imageDirection += vehicleEntry->restraint_image_id;
+
+    vehicle_sprite_paint(
+        session, vehicle, imageDirection, VehicleBoundboxes[vehicleEntry->draw_order][boundingBoxNum], z, vehicleEntry);
 }
 
 // 6D51DE
-static void vehicle_sprite_0_0(
+static void VehicleSpriteFlatUnbanked(
     paint_session& session, const Vehicle* vehicle, int32_t imageDirection, int32_t z,
     const rct_ride_entry_vehicle* vehicleEntry)
 {
-    vehicle_sprite_paint_6D51DE(session, vehicle, imageDirection, z, vehicleEntry);
+    if (vehicle->restraints_position >= 64 && (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_RESTRAINT_ANIMATION)
+        && (imageDirection & 7))
```

Comment:
shouldn't this conditional be inverted `!(imageDirection & 7)`

Reply:
or even better do `(imageDirection & 7) == 0` as it looks like its a condition for a singular rotation.

Reply:
It is testing if the vehicle is facing the axis. Good catch.

Reply:
Could also add a note with the following:

```
// Restraint animations are only drawn for vehicles that are in a cardinal direction (north, east, south, west)
```
This will be important if you ever want to get 64bit yaw working.